src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -741,6 +752,16 @@ private static void PartialMethodConstraintsChecks(SourceOrdinaryMethodSymbol de
                 var typeParameter1 = typeParameters1[i];
                 var typeParameter2 = typeParameters2[i];
 
+                // TODO: Should report on type parameter name differences?
```

Comment:
Sure, why not? However, it feels reasonable to do this at the same time as we check the parameter name difference. i.e. if any type parameter names or parameter names or parameter nullabilities differ, for example, report a single diagnostic on the implementation.

Reply:
@RikkiGibson I didn't do that because [LDM notes](https://github.com/dotnet/csharplang/blob/main/meetings/2020/LDM-2020-09-14.md#partial-method-signature-matching) didn't mention this at all. So I wasn't sure if it needs to be discussed in a later LDM first?

If it's okay to go ahead and do that directly in that PR, I'll.

Reply:
I feel that this language supports us reporting warnings on type parameter name differences:

> We will add a new warning wave warning for all other syntactic differences between partial method implementations.

I will go ahead and ping language design internally to see if there is any objection, but I'm fairly confident that this will be accepted.

Reply:
@RikkiGibson I did it in the last commit. In case there are any concerns not to do this, I'll revert.

Reply:
LDM has confirmed that type parameter name differences should also cause the warning.

Reply:
Thanks for confirming.